realsense2_camera/include/mavros_timestamp.h

```diff
@@ -0,0 +1,432 @@
+/*
+ Copyright (c) 2018, Michael Pantic, ASL, ETH Zurich, Switzerland
+                     Helen Oleynikova, ASL, ETH Zurich, Switzerland
+                     Zac Taylor, ASL, ETH Zurich, Switzerland
+
+ You can contact the author at <michael.pantic@mavt.ethz.ch>
+ All rights reserved.
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright
+ notice, this list of conditions and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright
+ notice, this list of conditions and the following disclaimer in the
+ documentation and/or other materials provided with the distribution.
+ * Neither the name of ETHZ-ASL nor the
+ names of its contributors may be used to endorse or promote products
+ derived from this software without specific prior written permission.
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL ETHZ-ASL BE LIABLE FOR ANY
+ DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef REALSENSE2_CAMERA_MAVROS_TIMESTAMP_H
+#define REALSENSE2_CAMERA_MAVROS_TIMESTAMP_H
+
+#include "ros/ros.h"
+#include <mavros_msgs/CamIMUStamp.h>
+#include <mavros_msgs/CommandTriggerControl.h>
+#include <sensor_msgs/Image.h>
+#include <boost/serialization/strong_typedef.hpp>
+#include <geometry_msgs/PointStamped.h>
+#include <mutex>
+#include <queue>
+#include <tuple>
+
+// Note on multi threading / locking:
+//      To avoid any confusion and non-defined state, this class locks a mutex for every function call
+//      that is not const. This is due to the fact that many callbacks can happen simultaneously, especially
+//      on callback based drivers such as the realsense. If not locked properly, this can lead to weird states.
+//      Note that all private functions assume that they are called in a locked context (no cascading locks).
+namespace mavros_trigger {
+
+enum trigger_state {
```

Comment:
I prefer enum class as they guard against a large number of accidental uses.

Reply:
Good point - changed it. Actually also prefer them, better scoping.